### PR TITLE
GeoJson implementation + Porting to Grafana 6.x

### PR DESCRIPTION
**Pull request by WT-JB**
fixed map rendering bug on grafana 6.x 
implemented geoJson support 
   -  in table, name field "geoJson" (this is case insensitive, so "geojson" or "geoJSON" are supported too, 
      makes no difference)
  -  geoJson objects will be shown on map in the same color as icons corresponding to them

### CODE REVIEW

File: src/worldmap.js [hunk 1]
```diff
@@ -115,41 +115,99 @@ export default class WorldMap {
...
-  createIcon(dataPoint) {
+  createGeoJson(dataPoint, geoJsonName) {
+    var geoColor = this.ctrl.panel.layersColors[dataPoint.type];
+    if (geoColor === 'lightred') {
+        geoColor = '#FF9898';
+    } else if (geoColor === 'darkpurple') {
+        geoColor = '#6813B2';
+    } else if (geoColor === 'black') {
+        geoColor = '#404040';
+    } else if (geoColor === null && this.ctrl.panel.layersIcons[dataPoint.type] !== null) {
```
Comment (taliaga) [inline src/worldmap.js]: what is this case for again?

File: src/worldmap.js [hunk 2]
```diff
@@ -115,41 +115,99 @@ export default class WorldMap {
...
   }
 
-  createIcon(dataPoint) {
+  createGeoJson(dataPoint, geoJsonName) {
+    var geoColor = this.ctrl.panel.layersColors[dataPoint.type];
+    if (geoColor === 'lightred') {
+        geoColor = '#FF9898';
+    } else if (geoColor === 'darkpurple') {
+        geoColor = '#6813B2';
+    } else if (geoColor === 'black') {
```
Comment (taliaga) [inline src/worldmap.js]: curiosity: why only these colours?

File: src/worldmap.js [hunk 3]
```diff
@@ -115,41 +115,99 @@ export default class WorldMap {
...
+    if(typeof dataPoint[geoJsonName] === 'object') {
+        retVal = L.geoJSON(dataPoint[geoJsonName], {
+            style: myStyle
+        });
+    } else {
+        retVal = L.geoJSON(JSON.parse(dataPoint[geoJsonName]), {
+          style: myStyle
+        });
+    }
+    var dataInfoWithoutGeoJson = JSON.parse(JSON.stringify(dataPoint)); //creates clone of json
```
Comment (taliaga) [inline src/worldmap.js]: why is the clone needed again? is the "delete" if L185 causing a data loss? could we instead adjust the method retrieving the info to not display the geojson?

**Comment by taliaga** [comment]
@WT-JB , Also, I tested with a geoshape of type "Point" and I noticed an extra blue non-iconed marker gets created together. Have you tested that case?. See https://github.com/orchestracities/grafana-map-plugin/blob/1acaa67d2df4d6997cf6fc263de59fa524f99deb/create-table.sh#L46 for example.

File: src/worldmap.js (hunk 1 above)
Comment (WhitesteinDevelopment) [reply src/worldmap.js]: this piece of code is no more relevant, I have removed it

File: src/worldmap.js (hunk 2 above)
Comment (WhitesteinDevelopment) [reply src/worldmap.js]: this piece of code is no more relevant, I have removed it

File: src/worldmap.js (hunk 3 above)
Comment (WhitesteinDevelopment) [reply src/worldmap.js]: I have re-implemented as you wanted (no cloning, just skip the geojson info in popup).

**Pull request merged**